### PR TITLE
Added option to invert thumbnails in dark mode

### PR DESCRIPTION
**Pull request by BolkoSchreiber**
see issue #604

### CODE REVIEW

**Comment by shamoon** [comment]
Nice. IMHO I think this should default to true since that’s what it currently is (and is probably more in keeping with dark mode)

**Comment by shamoon** [comment]
Also could make css more DRY by just using an `inverted` class and rules for `.doc-img.inverted`

(sorry I’m on my phone)

**Comment by BolkoSchreiber** [comment]
you are right - that default value was a bit selfish :)

About the css: my experience with css is "next to nothing" - but I'll do what i can

**Comment by shamoon** [comment]
> About the css: my experience with css is "next to nothing"

I can suggest specifics

File: src-ui/src/app/components/document-list/document-card-large/document-card-large.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="card mb-3 shadow-sm" [class.card-selected]="selected" [class.document-card]="selectable">
   <div class="row no-gutters">
     <div class="col-md-2 d-none d-lg-block doc-img-background rounded-left" [class.doc-img-background-selected]="selected" (click)="this.toggleSelected.emit($event)">
-      <img [src]="getThumbUrl()" class="card-img doc-img border-right rounded-left">
+      <img [src]="getThumbUrl()" class="card-img {{ getIsThumbInverted() ? 'doc-img-inverted' : 'doc-img' }} border-right rounded-left">
```
Comment (shamoon) [inline src-ui/src/app/components/document-list/document-card-large/document-card-large.component.html:4]: ``

File: src-ui/src/app/components/document-list/document-card-large/document-card-large.component.scss [hunk 2]
```diff
@@ -4,7 +4,7 @@
   overflow-wrap: anywhere;
 }
 
-.doc-img {
+.doc-img, .doc-img-inverted {
```
Comment (shamoon) [inline src-ui/src/app/components/document-list/document-card-large/document-card-large.component.scss:7]: `.doc-img {`

File: src-ui/src/app/components/document-list/document-card-small/document-card-small.component.scss [hunk 3]
```diff
@@ -1,6 +1,6 @@
 @import "/src/theme";
 
-.doc-img {
+.doc-img, .doc-img-inverted {
```
Comment (shamoon) [inline src-ui/src/app/components/document-list/document-card-small/document-card-small.component.scss:3]: `.doc-img {`

File: src-ui/src/app/components/document-list/document-card-small/document-card-small.component.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <div class="col p-2 h-100">
   <div class="card h-100 shadow-sm document-card" [class.card-selected]="selected">
     <div class="border-bottom doc-img-container" [class.doc-img-background-selected]="selected" (click)="this.toggleSelected.emit($event)">
-      <img class="card-img doc-img rounded-top" [src]="getThumbUrl()">
+      <img class="card-img {{ getIsThumbInverted() ? 'doc-img-inverted' : 'doc-img' }} rounded-top" [src]="getThumbUrl()">
```
Comment (shamoon) [inline src-ui/src/app/components/document-list/document-card-small/document-card-small.component.html:4]: ``

File: src-ui/src/theme_dark.scss [hunk 5]
```diff
@@ -148,7 +158,7 @@ $border-color-dark-mode: #47494f;
     }
   }
 
-  .card-selected .doc-img {
+  .card-selected .doc-img .doc-img-inverted {
```
Comment (shamoon) [inline src-ui/src/theme_dark.scss:161]: `.card-selected .doc-img {`

File: src-ui/src/theme_dark.scss [hunk 6]
```diff
@@ -138,6 +138,16 @@ $border-color-dark-mode: #47494f;
...
+    mix-blend-mode: normal;
+    border-radius: 0;
+    border-color: $bg-dark-mode;
+
+    &.border-right {
+      border-right: none !important;
+    }
+  }
+
+  .doc-img-inverted {
```
Comment (shamoon) [inline src-ui/src/theme_dark.scss:150]: `.doc-img.inverted {`

(or if you nest this inside .doc-img you can use &.inverted )

**Review by shamoon — CHANGES_REQUESTED** [review]
Please note Im not testing these changes on my machine at the moment in case theres a typo or something but I think you get the idea.

File: src-ui/src/theme_dark.scss [hunk 7]
```diff
@@ -138,6 +138,16 @@ $border-color-dark-mode: #47494f;
...
     mix-blend-mode: normal;
     filter: invert(95%) hue-rotate(180deg);
     border-radius: 0;
```
Comment (shamoon) [inline src-ui/src/theme_dark.scss:153]: with other changes this becomes just
`filter: invert(95%) hue-rotate(180deg);`
(other lines become redundant)

**Comment by BolkoSchreiber** [comment]
Thank you! I think I got it now

**Review by shamoon — APPROVED** [review]

**Comment by jonaswinkler** [comment]
Seem's you've got it already sorted out! Great.

Maybe it's still a good idea to have the bright thumbnails edited slightly in dark mode (maybe `invert(5%)' or something), just so they're a little easier on the eyes.

**Comment by BolkoSchreiber** [comment]
ok ... pick one:

0%:
grafik

5%:
grafik

10% (my favorite):
grafik

15%:
grafik


for the sake of color correctness, I need to add hue-rotate with a fraction of 180°, right? (invert 5% -> 9deg hue-rotate)

**Comment by jonaswinkler** [comment]
Choose whatever you think looks best!

**Comment by BolkoSchreiber** [comment]
Alright. 10% it is.

Thank you for your input.

**Comment by jonaswinkler** [comment]
I've merged this into `dev` manually now, and the PR will show up as merged with the next release.

Next time, please make functional changes on the `dev` branch. `master` always has the code for the latest release.

**Pull request merged**